_posts/2020-01-01-SVGD.markdown

```diff
@@ -0,0 +1,405 @@
+
+          3. SVGD still seems to collapse in high-dimensional spaces, leading to exciting new research in <a href="https://arxiv.org/abs/1902.03394">why this occurs</a> and <a href="https://arxiv.org/abs/1711.04425">ideas on how to get around it</a>.
+    </p>
+  </details>
+
+  2. Prove that the discrepancy in Equation 3 of the Stein Variational Gradient Descent Paper only equals 0 when \(p\) and \(q\) are equal.
+
+  <details><summary>Solution</summary>
+    <p>
+      Recall the operator definition of Stein's Identity:
+
+      $$ \mathbf{E}_p[\mathcal{A}_pf(x)] = 0$$
+
+      If \( p \neq q \), we get \( \mathbf{E}_q[\mathcal{A}_pf(x)] \) for some choice of function \( f \).
+
+      We can expand this to:
+
+      $$\mathbf{E}_q[\mathcal{A}_pf(x)] - \mathbf{E}_q[\mathcal{A}_qf(x)]$$
+
+      Recalling the full definition of the operator:
+
+      $$\mathcal{A}_pf(x) = \mathbf{E}_p[s_p(x)f(x) + \nabla_x f(x)] = 0$$
+
+      where score function \( s_p(x) \) is just \( \nabla_x \log p(x) \), we are left with
+
+      $$\mathbf{E}_q[(s_p(x) - s_q(x))f(x)]$$
+
+      This means unless \(p = q \rightarrow s_p(x) = s_q(x) \; \forall x \in \mathcal{X} \), we can always find some function \(f\) for which the above quantity is nonzero.
+
+    </p>
+  </details>
+
+  3. Implement SVGD in your favorite language (see the notes for links to different implementations). Then, let’s take a look at the role of the kernel in SVGD: (a) Remove the repulsive kernel term and observe how particles collapse to modes. (b) Remove the kernel’s contribution in the first term. What happens?	
+
+<br />
+
+# 5 Stein in Reinforcement Learning
+  **Motivation**: One of the most exciting use cases of SVGD is in reinforcement learning, due to its connection to maximum entropy reinforcement learning. This week, we study two key techniques in reinforcement learning that use SVGD as the underlying mechanism. In reinforcement learning, the target distribution is not known, so we derive gradient updates to our parameters using policy gradients. As we derive the gradient estimators in the maximum-entropy framework of reinforcement learning, we will start to see what benefits SVGD-based methods have. In particular, we will focus on the explore-exploit tradeoff, as well as normalization constants for intractable distributions, and see how SVGD helps us get around complicated problems regarding both. 
+
+  **Topics**:
+
+  1. Reinforcement Learning
+  2. Explore vs. Exploit
+  3. Maximum Entropy Reinforcement Learning
+
+  **Notes**:  In this class, with notes in [Colab](https://colab.research.google.com/drive/178X8BgGrUmPaRTLulL_ETUKaBf-MfrgS) or [PDF](/assets/svgd_notes/week06.pdf), we look at the application area of _Reinforcement Learning_, and see how the diversity induced by SVGD (and its connection to maxmimum entropy reinforcement learning) generates strongly-exploring policies. 
```

Comment:
"we look at the application area of _Reinforcement Learning_, and see..." --> "we look at reinforcement learning and see..."